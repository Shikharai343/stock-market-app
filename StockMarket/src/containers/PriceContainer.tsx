import React, {Component, Fragment} from 'react';
import {Button, StyleSheet, TextInput, View} from 'react-native';

interface IProps {
    navigation: {
        getParam: (query: string) => any;
    };
}

class PriceContainer extends Component <IProps> {
    constructor(props: IProps) {
        super(props);
        this.state = {
            text: this.props.navigation.getParam('itemPrice'),
        };
    }

    render() {
        return (
            <Fragment>
                <TextInput
                    style={styles.inputStyle}
                    onChangeText={(text)=> this.setState({text})}
                    value={this.state.text}
                    keyboardType={'numeric'}
                />
                <View style={styles.buttons}>
                    <Button
                        title='Save'
                        onPress={() => this.props.navigation.getParam('updateState')(this.state.text)}
                    />
                    <Button
                        title='Delete'
                        color='#b8092c'
                        onPress={this.props.navigation.getParam('deleteData')} />
                </View>
            </Fragment>
        );
    }
}

const styles = StyleSheet.create({
    inputStyle: {
        height: 30,
        width: 150,
        borderColor: 'black',
        borderWidth: 2,
        borderRadius: 5,
        marginTop: '5%',
        marginBottom: '5%',
        textAlign: 'center'
    },
    buttons: {
        flexDirection: 'row',
    }
});

export default PriceContainer;
