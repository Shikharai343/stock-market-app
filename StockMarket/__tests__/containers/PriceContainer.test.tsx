import React from 'react';
import { shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import PriceContainer from '../../src/containers/PriceContainer';
import {Button, TextInput} from 'react-native';

const mockNavigate = jest.fn().mockImplementation(() => 'okay');
const mockNavigation = {
    navigate: mockNavigate,
    navigation: {
        getParam: mockNavigate,
    },
};
let wrapper: any;

describe('Snapshot testing', () => {
    wrapper = shallow(<PriceContainer  navigation={mockNavigation.navigation}/>);

    it('should create a snapshot of the StockPrice Screen', () => {
        const tree = renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should call with param', () => {
        expect(mockNavigation.navigation.getParam).toHaveBeenCalledWith('itemPrice');
        expect(mockNavigation.navigation.getParam).toHaveBeenCalledWith('deleteData');
    });

    it('should return button length', () => {
       expect( wrapper.find(Button)).toHaveLength(2);
    })

    it('should change state when price entered', () => {
        wrapper.find(TextInput).first().prop('onChangeText')('123');
        expect(wrapper.state('text')).toBe('123');
    })
});
