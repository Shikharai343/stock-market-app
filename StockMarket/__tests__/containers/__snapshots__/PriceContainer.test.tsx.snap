// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing should create a snapshot of the StockPrice Screen 1`] = `
Array [
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "black",
        "borderRadius": 5,
        "borderWidth": 2,
        "height": 30,
        "marginBottom": "5%",
        "marginTop": "5%",
        "textAlign": "center",
        "width": 150,
      }
    }
    underlineColorAndroid="transparent"
    value="okay"
  />,
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="saveButton"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Save
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "width": 10,
        }
      }
    />
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#b8092c",
              },
            ]
          }
        >
          Delete
        </Text>
      </View>
    </View>
  </View>,
]
`;
